indico/web/client/js/react/components/ICSCalendarLink.jsx

```diff
@@ -119,15 +127,23 @@ export default function ICSCalendarLink({endpoint, params, renderButton, popupPo
     dispatch({type: 'CLOSE'});
   };
 
-  const handleSetOption = async (key, extraParams) => {
+  const _handleSetOption = async (key, extraParams) => {
+    if (popupState.source) {
+      popupState.source.cancel();
+    }
+    const source = indicoAxios.CancelToken.source();
+    dispatch({type: 'OPEN', key, source});
+    const url = await fetchURL(extraParams, source);
+    dispatch({type: 'LOADED', url});
+  };
+
+  const handleSetOption = (key, extraParams) => {
     if (!popupState.open || popupState.key !== key) {
-      if (popupState.source) {
-        popupState.source.cancel();
-      }
-      const source = indicoAxios.CancelToken.source();
-      dispatch({type: 'OPEN', key, source});
-      const url = await fetchURL(extraParams, source);
-      dispatch({type: 'LOADED', url});
+      extraParams = {
+        ...extraParams,
+        ...(exportEventSeries && {export_event_series: exportEventSeries}),
+      };
```

Comment:
Wouldn't something like this be more readable?

```js
if (exportEventSeries) {
  extraParams = {...extraParams, export_event_series: true};
}
```

(I guess you want the assignment to avoid mutating the original `extraParms`?)

Reply:
Yup, much better with the `if ..`